spec/models/geo_entity_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.describe GeoEntity, type: :model do
+  include_context 'geo_entity_stat_setup'
+
+  before do
+    habitats.each do |habitat|
+      countries.each do |country|
+        FactoryBot.create(:geo_entity_stat, habitat: habitat, geo_entity: country)
+      end
+    end
+  end
+
+  let(:geo_entity) { GeoEntity.last }
+
+  describe '#count_species' do
+    it 'calls the relevant method on the Species model' do
```

Comment:
Took me a while to work out what was being tested here. Would it be against convention for the description to be 'calls count_species on the Species model when calling count_species on itself'. 

It sounds a bit over the top, but I think I was confused because of the two methods being named the same and I missed for a while that they were defined on different models... Not a big deal, just might save time for people looking at it in the future. Ignore if you think that makes it more confusing!

Reply:
The way this method works is a bit convoluted in that it's calling a method that exists on the Species model, which shouldn't really be happening as ideally that could all be handled within the Species model itself. I might look at removing this spec (and refactor that particular method along with any method calls to it) in the future.